packages/editor/src/components/dialogs/PublishModal.tsx

```diff
@@ -0,0 +1,271 @@
+provide for limited attribution for the Original Developer. In addition, 
+Exhibit A has been modified to be consistent with Exhibit B.
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+The Original Code is Ethereal Engine.
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Ethereal Engine team.
+All portions of the code written by the Ethereal Engine team are Copyright © 2021-2023 
+Ethereal Engine. All Rights Reserved.
+*/
+
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+
+import { NotificationService } from '@etherealengine/client-core/src/common/services/NotificationService'
+import { PopoverState } from '@etherealengine/client-core/src/common/services/PopoverState'
+import {
+  LocationData,
+  LocationID,
+  locationPath,
+  LocationType,
+  staticResourcePath
+} from '@etherealengine/common/src/schema.type.module'
+import { useHookstate } from '@etherealengine/hyperflux'
+import { useFind, useMutation } from '@etherealengine/spatial/src/common/functions/FeathersHooks'
+import Button from '@etherealengine/ui/src/primitives/tailwind/Button'
+import Input from '@etherealengine/ui/src/primitives/tailwind/Input'
+import LoadingView from '@etherealengine/ui/src/primitives/tailwind/LoadingView'
+import { ModalHeader } from '@etherealengine/ui/src/primitives/tailwind/Modal'
+import Select from '@etherealengine/ui/src/primitives/tailwind/Select'
+import Toggle from '@etherealengine/ui/src/primitives/tailwind/Toggle'
+import { HiLink } from 'react-icons/hi2'
+
+const getDefaultErrors = () => ({
+  name: '',
+  maxUsers: '',
+  scene: '',
+  serverError: ''
+})
+
+const locationTypeOptions = [
+  { label: 'Private', value: 'private' },
+  { label: 'Public', value: 'public' },
+  { label: 'Showroom', value: 'showroom' }
+]
+
+export default function PublishSceneModal({ location, sceneID }: { location?: LocationType; sceneID: string | null }) {
+  const { t } = useTranslation()
+
+  const locationMutation = useMutation(locationPath)
+
+  const submitLoading = useHookstate(false)
+  const errors = useHookstate(getDefaultErrors())
+
+  const name = useHookstate(location?.name || '')
+  const maxUsers = useHookstate(location?.maxUsersPerInstance || 20)
+
+  const scene = useHookstate((location ? location.sceneId : sceneID) ?? '')
+  const videoEnabled = useHookstate<boolean>(location?.locationSetting.videoEnabled || true)
+  const audioEnabled = useHookstate<boolean>(location?.locationSetting.audioEnabled || true)
+  const screenSharingEnabled = useHookstate<boolean>(location?.locationSetting.screenSharingEnabled || true)
+  const locationType = useHookstate(location?.locationSetting.locationType || 'public')
+
+  const scenes = useFind(staticResourcePath, {
+    query: {
+      paginate: false
```

Comment:
This needs to have `type: 'scene'`

Do this fix for admin panel dialog aswell.

Reply:
done